ykllvmwrap/src/ykllvmwrap.cc

```diff
@@ -437,20 +437,17 @@ extern "C" void *__ykllvmwrap_irtrace_compile(char *FuncNames[], size_t BBs[],
               errx(EXIT_FAILURE, "Non-const global variable %s",
                    OldGV->getName().data());
             }
-          } else {
-            if (OpTy->isIntegerTy()) {
-              // Value is an integer constant, so leave it as is.
-              // FIXME Extend this with other types as needed to get new
-              // tests to run. Ultimately, find a better way to do this.
-              VMap[Op] = Op;
-              continue;
-            } else {
-              // Value is not a stack allocation or constant, so make a dummy
-              // default value.
-              // FIXME fails for meta-data types (when you build with -g).
-              Value *NullVal = Constant::getNullValue(OpTy);
-              VMap[Op] = NullVal;
-            }
+          } else if (isa<Constant>(Op)) {
+            // The operand is a constant, so leave it as is.
+            VMap[Op] = Op;
+            continue;
+          } else if (Op == ThreadTracer) {
```

Comment:
I *think* `ThreadTrace` can be `null` at this point. Does that cause issues?

Reply:
It shouldn't be, as we only flip `Tracing` on when we see the instruction that generates `ThreadTracer`.

Reply:
Do you mean `ThreadTracer` can't be null? If so, maybe add a (debug) assert after the `for` loop to check this?

Reply:
ThreadTracer cannot be null if `Tracing` is set.

When we see the call that turns on tracing:
```                                                                                                                                                                                                      
        if (CF->getName() == YKTRACE_START) {                                                                                                                                                                      
          ThreadTracer = &*I;                                                                                                                                                                                      
          Tracing = true;        
```

It occurs to me that we don't need the `Tracing` flag, as we can just check the nullness of this pointer. Would that be preferential / clearer?

Reply:
I think we should keep `Tracing` as it's much clearer what it's for than comparing to some pointer.

Reply:
Having two variables implicitly track the same thing seems like it's more confusing, at least to me.

Reply:
@ptersilie How about if we renamed `ThreadTracer` to `StartTracingInstr`? That way it makes some sense in both contexts. We are tracing if we've seen the `StartTracingInstr` and the value produced is the `ThreadTracer`.

Reply:
I'm happy with that.

Reply:
Have a look at 3708c9f2da0. I'm on the fence as to whether it's better or not.

Reply:
This reads better to me.